### DISCUSSION TITLE
cannot acces Port from outside of docker host

### DISCUSSION THREAD

**Discussion by Torkolis** [dotnet/dotnet-docker#3856] [Q&A]
Hi, I have built the image with the dotnet-docker/samples/aspnetapp/Dockerfile
and got my image and also ran it with 
`docker run --name myapp --rm -itd -p 8000:80 myapp`
and then I have an issue that connections from outside of docker host VM seem to be blocked by a firewall or something.
curl from the docker host (which is a hyperV VM with ubuntu 22.04 OS) itself is fine but curl from my Notebook is blocked:
```
curl -v bdockervm.mshome.net:8000
*   Trying 172.17.101.195:8000...
* connect to 172.17.101.195 port 8000 failed: Timed out
* Failed to connect to bdockervm.mshome.net port 8000 after 21032 ms: Timed out
* Closing connection 0
curl: (28) Failed to connect to bdockervm.mshome.net port 8000 after 21032 ms: Timed out
```
I have never seen this issue with other containers, usually once a container is up and port mapped I can access it with browser from my Notebook without any problems whatsoever. So I assume it is something special about this image  mcr.microsoft.com/dotnet/aspnet:6.0

**Comment by richlander** [comment]
Here's my experience:

```bash
% docker build --pull -t aspnetapp .
% docker run -d --name myapp --rm -p 8000:80 aspnetapp  
% curl localhost:8000/Environment
{"runtimeVersion":".NET 6.0.6","osVersion":"Linux 5.10.104-linuxkit #1 SMP PREEMPT Thu Mar 17 17:05:54 UTC 2022","osArchitecture":"Arm64","processorCount":4,"totalAvailableMemoryBytes":4108656640,"memoryLimit":0,"memoryUsage":0}
```

I always thought of `it` and `d` as mutually exclusive. I never use them together. What's you thinking on that?

**Reply by Torkolis** [reply]
from my exprience there is no issue with `it` and `d`, I use them together very often (even though I'm not sure why). And as I wrote `localhost` works fine, I just have an issue from another computer which can reach the docker host over the network.

**Reply by richlander** [reply]
`it` relate to setting up a console. `d` is about disconnected. To my mind `d` cancels out `it`. Happy to be informed otherwise.

I had a problem with granting access over the network at point. I cannot remember right now what the solution to that was.

**Reply by Torkolis** [reply]
> I had a problem with granting access over the network at point. I cannot remember right now what the solution to that was.

That is probably my issue now.

**Reply by Torkolis** [reply]
Ok now it works strangely. I have not done anything different from yesterday, but now it works, even with `it` and `d` together.

**Reply by richlander** [reply]
Great. Looks like this is resolved.
